File: src/functions/GetAllPosts.ts
import { response } from '@libs/api-gateway';
import { PostService } from '../libs/post.service';
import { APIGatewayEvent, APIGatewayProxyHandler } from "aws-lambda";

export const handler: APIGatewayProxyHandler = async (
  event: APIGatewayEvent
): Promise<any> => {

  const postService: PostService = new PostService();
  const posts = await postService.getAllPosts();

  return response(200, posts);
};

File: src/functions/GetAllPosts.go
package functions

import (
	"context"
	"fmt"
	"net/http"

	"github.com/aws/aws-lambda-go/events"

	"blog-cms-v2/src/libs"
)

func GetAllPosts(ctx context.Context, _ events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
	postService := libs.NewPostService()
	posts, err := postService.GetAllPosts(ctx)
	if err != nil {
		return libs.ErrResponse(http.StatusInternalServerError, fmt.Errorf("fetching posts: %w", err)), nil
	}
	return libs.Response(http.StatusOK, posts), nil
}
